### PR TITLE
[SHELL32_APITEST] Strengthen and improve ShellHook testcase

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

- Add tests to find the missing parameters against `HSHELL_WINDOWCREATED` conditions.
- Improve code readability.

JIRA issue: [CORE-15669](https://jira.reactos.org/browse/CORE-15669)

## Proposed changes

- Simplify the code.
- Don't use magic numbers.
- Add `STYLE_2` for tests.

## TODO

- [x] Screenshots.

### CODE REVIEW

**Comment by katahiromz** [comment]
WinXP:
WinXP
Successful.

Win2k3:
Win2k3
Successful.

Win10:
Win10
Successful.

ReactOS:
ReactOS
We found the missing parameters.

**Comment by katahiromz** [comment]
Check time.

**Comment by katahiromz** [comment]
I think the non-`WS_VISIBLE` cases are trivial, so I will omit them.

**Comment by katahiromz** [comment]
WinXP:
WinXP
Win2k3:
Win2k3
Win10:
Win10
ReactOS:
ReactOS

**Comment by katahiromz** [comment]
Check time. I will merge 3 days later.

File: modules/rostests/apitests/shell32/ShellHook.cpp [hunk 1]
```diff
@@ -357,65 +305,43 @@ WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
-                    dwFlags |= (1 << 5);
-                if (owner_exstyle & WS_EX_APPWINDOW)
-                    dwFlags |= (1 << 6);
-                s_dwFlags = dwFlags;
+                ++s_nWindowCreatedCount;
                 break;
         }
     }
+#define ID_IGNITION 1000
+#define ID_BURNING 2000
```
Comment (HBelusca) [inline modules/rostests/apitests/shell32/ShellHook.cpp:318]: 🚀

File: modules/rostests/apitests/shell32/ShellHook.cpp [hunk 2]
```diff
@@ -357,65 +305,43 @@ WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
...
                     break;
-                style = (LONG)GetWindowLongPtrW(s_hwndTarget, GWL_STYLE);
-                exstyle = (LONG)GetWindowLongPtrW(s_hwndTarget, GWL_EXSTYLE);
-                if (style & WS_CHILD)
-                    hwndOwner = GetParent(s_hwndTarget);
-                else
-                    hwndOwner = GetWindow(s_hwndTarget, GW_OWNER);
-                owner_style = (LONG)GetWindowLongPtrW(hwndOwner, GWL_STYLE);
-                owner_exstyle = (LONG)GetWindowLongPtrW(hwndOwner, GWL_EXSTYLE);
-                dwFlags = (1 << 0);
```
Comment (HBelusca) [inline modules/rostests/apitests/shell32/ShellHook.cpp:376]: So if I understand correctly, using window styles for comparisons was not a reliable test?

Comment (katahiromz) [reply modules/rostests/apitests/shell32/ShellHook.cpp:376]: This was excessive tests.

**Pull request merged**